src/test/java/de/tum/in/www1/artemis/TextExerciseIntegrationTest.java

```diff
@@ -107,6 +110,26 @@ public void deleteExamTextExercise() throws Exception {
         assertThat(textExerciseRepository.findAll()).isEmpty();
     }
 
+    @Test
+    @WithMockUser(value = "tutor1", roles = "INSTRUCTOR")
+    public void deleteTextExercise_notFound() throws Exception {
+        TextExercise textExercise = new TextExercise();
+        textExercise.setId(114213211L);
+
+        request.delete("/api/text-exercises/" + textExercise.getId(), HttpStatus.NOT_FOUND);
+    }
+
+    @Test
+    @WithMockUser(value = "instructor1", roles = "INSTRUCTOR")
+    public void deleteTextExercise_isNotAtLeastAnInstructor_forbidden() throws Exception {
```

Comment:
I think the combination of test name and used `@WithMockUser` is a bit confusing here. I would assume that something is forbidden if you are not at least instructor but at the same time, the test user _is_ instructor. I figured, that this is because the test does not check the actual role but rather the group name, which is 'instructor' for the given user but set to 'test' for the course making the request returning `HttpStatus.Forbidden`, which works fine. But I would rather call the method `deleteTextExercise_isWrongInsructorGroup_forbidden()` then or check for the actual role (and keep the name) -> i.e. just using `@WithMockUser(value = "tutor1", roles = "TA")` and remove the statement where you set the instructor group name. I tried it, it should work :)

Reply:
TA does not have access to edit/delete the exercise. So, if I change the mockuser as TA, it will now reach this endpoint, so this case will not be covered.

Reply:
same applies to: `testGetPlagiarismResult_isNotAtLeastInstructor_forbidden()`,
`testCheckPlagiarism_isNotAtLeastInstructor_forbidden()`, `getTextExercise_isNotAtleastTeachingAssistant_forbidden()`, `getAllTextExercisesForCourse_isNotAtLeastTeachingAssistantInCourse_forbidden()`, `updateTextExercise_isNotAtLeastAnInstructor_forbidden()`,
`createTextExercise_isNotAtLeastAnInstructor_forbidden()`

Reply:
a yes! I got your point ;) Then I would change it to `deleteTextExercise_isWrongInsructorGroup_forbidden()` or stick to the pattern `*_isNotAtLeastXInCourse_forbidden()` where the _inCourse_ suggests that the role might be correct, but the group name is not for this course.

Reply:
yeah I am not really fan of the current name for these cases. I like the your second suggestion, I will change like that..